elastic/kibana
# [Fleet] Make default integration install explicit

## Summary

Making default integration install explicit
Fixes https://github.com/elastic/kibana/issues/108456

Manual test cases:

Initial kibana startup:
1. Navigate to Fleet, verify that /setup has finished (should happen quickly)
2. verify that no default policies or integrations exist (except for cloud fleet server policy, see below)

Add integration flow:
1. go to Integrations / Apache / Add integration
2. verify that UI offers to create a new policy
3. when submitting, verify that agent policy is being created with system integration, and elastic agent integration is installed as well (first time takes long, about 2 mins)
4. try again to add integration, should see now the tabbed view New hosts / Existing hosts
5. test both paths: create new policy and use existing policy

Fleet Server policy flow on-prem:
1. Go to Fleet tab
2. verify that in fleet server instructions there is a create button to add fleet server policy
3. click create button
4. verify that fleet server policy is created, UI should show a dropdown instead of create button now 
5. verify that fleet server policy has a fleet server integration 
6. verify policy has system integration if system monitoring checkbox was left checked as default
6. elastic agent integration should be installed as well if monitoring was enabled on the policy
7. generated fleet server command should include the newly created policy id

Add agent flow (supposing Fleet Server is installed):
1. go to Fleet / Agents / Add agent
2. on both Enroll in Fleet / Run standalone tab, see option to create Agent policy (when no policy exists, except fleet server policy)
3. click on Create button, verify that default policy is created with system integration, and elastic agent integration is installed
4. verify that generated enroll command (or policy config yml) includes newly created policy id
5. verify that now agent policy dropdown shows up with pre-selected agent policy
6. create one more agent policy, and verify that dropdown has no selection by default

Cloud (depends on change in cloud stackpack)
1. Verify that Elastic Cloud agent policy is created by default with fleet server integration
2. non fleet server flows are the same as on prem (no default policy, system, elastic agent integration initially)

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
